### PR TITLE
In CI use runner's JDK17 by default

### LINKED ISSUE #14141: Random build failure because Java 11 gets picked
`ci / maven-checks (17) (pull_request)`   https://github.com/trinodb/trino/actions/runs/3058962341/jobs/4935779621 printed
```
2022-09-15T08:39:42.0824486Z + ./mvnw clean verify -B --strict-checksums -V -T C1 -DskipTests -P ci -pl !:trino-server-rpm
2022-09-15T08:39:46.6413510Z Apache Maven 3.8.4 (9b656c72d54e5bacbed989b64718c159fe39b537)
2022-09-15T08:39:46.6414739Z Maven home: /home/runner/.m2/wrapper/dists/apache-maven-3.8.4-bin/5sjqtqkfjmakil1qhdtgrnjjdc/apache-maven-3.8.4
2022-09-15T08:39:46.6415511Z Java version: 11.0.16.1, vendor: Eclipse Adoptium, runtime: /usr/lib/jvm/temurin-11-jdk-amd64
....
2022-09-15T08:42:44.5858563Z [WARNING] Rule 5: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
2022-09-15T08:42:44.5860076Z Detected JDK Version: 11.0.16-1 is not in the allowed range 17.0.4.
....
```

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

Fixes #14141. This makes me wonder if we should be downloading the JDK at all. We could replace `actions/setup-java` with `actions/cache` (since we do use it implicitly for the local Maven repository).

## Release notes

(x) This is not user-visible and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by hashhar — CHANGES_REQUESTED** [review]
This removes whatever little reproducibility we had. I'd argue to do opposite - explicitly unset JVM_HOME before running setup-java to make sure setup-java failures (even if silent) actually lead to failure.

**Comment by nineinchnick** [comment]
I agree that turning silent failures into visible ones is much better. I updated this PR.

**Comment by findepi** [comment]
> explicitly unset JVM_HOME before running setup-java to make sure setup-java failures (even if silent) actually lead to failure.

i am not sure this is sufficient

do you know why  setup-java failures can be silent?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -51,6 +51,9 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0 # checkout all commits, as the build result depends on `git describe` equivalent
+      # Unset the default Java in case actions/setup-java silently fails
```
Comment (findepi) [inline .github/workflows/ci.yml:54]: Should we report this to GH?

Comment (nineinchnick) [reply .github/workflows/ci.yml:54]: Yes, we should open an issue in https://github.com/actions/setup-java but I'm not sure how to describe it.

**Comment by nineinchnick** [comment]
> do you know why setup-java failures can be silent?

No, I have no idea. If there would be some network issue I'd expect more errors to get logged, instead of the action just terminating.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (findepi) [reply .github/workflows/ci.yml:54]: i guess https://github.com/trinodb/trino/issues/14141#issuecomment-1248042941 is all we know so far.

**Comment by nineinchnick** [comment]
Looks like this one lost traction. I'll close it for now and maybe we could get back to it once either #14865 or #12817 gets merged.

**Pull request closed without merging**